src/plugins/action_server/action_server_impl.cpp

```diff
@@ -0,0 +1,312 @@
+            return ActionServer::FlightMode::Land;
+        case SystemImpl::FlightMode::Offboard:
+            return ActionServer::FlightMode::Offboard;
+        case SystemImpl::FlightMode::FollowMe:
+            return ActionServer::FlightMode::FollowMe;
+        case SystemImpl::FlightMode::Manual:
+            return ActionServer::FlightMode::Manual;
+        case SystemImpl::FlightMode::Posctl:
+            return ActionServer::FlightMode::Posctl;
+        case SystemImpl::FlightMode::Altctl:
+            return ActionServer::FlightMode::Altctl;
+        case SystemImpl::FlightMode::Acro:
+            return ActionServer::FlightMode::Acro;
+        case SystemImpl::FlightMode::Stabilized:
+            return ActionServer::FlightMode::Stabilized;
+        default:
+            return ActionServer::FlightMode::Unknown;
+    }
+}
+
+ActionServerImpl::ActionServerImpl(System& system) : PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ActionServerImpl::ActionServerImpl(std::shared_ptr<System> system) : PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ActionServerImpl::~ActionServerImpl()
+{
+    _parent->unregister_plugin(this);
+}
+
+void ActionServerImpl::init() {
+        // Arming / Disarm / Kill
+        _parent->register_mavlink_command_handler(
+        MAV_CMD_COMPONENT_ARM_DISARM,
+        [this](const MavlinkCommandReceiver::CommandLong& command) {
+            ActionServer::ArmDisarm armDisarm{
+                command.params.param1 == 1,
+                command.params.param2 == 21196};
+
+            // Check arm states - Ugly.
+            uint8_t request_ack = MAV_RESULT_UNSUPPORTED;
+            bool force = armDisarm.force;
+            if (armDisarm.arm) {
+                request_ack = (_armable || (force && _force_armable)) ?
+                                  MAV_RESULT::MAV_RESULT_ACCEPTED :
+                                  MAV_RESULT_TEMPORARILY_REJECTED;
+            } else {
+                request_ack = (_disarmable || (force && _force_disarmable)) ?
+                                  MAV_RESULT::MAV_RESULT_ACCEPTED :
+                                  MAV_RESULT_TEMPORARILY_REJECTED;
+            }
+
+            if (request_ack == MAV_RESULT::MAV_RESULT_ACCEPTED) {
+                _parent->set_server_armed(armDisarm.arm == 1);
+            }
+
+            auto result = (request_ack == MAV_RESULT::MAV_RESULT_ACCEPTED) ?
+                              ActionServer::Result::Success :
+                              ActionServer::Result::CommandDenied;
+
+            _parent->call_user_callback(
+                [this, armDisarm, result]() { _arm_disarm_callback(result, armDisarm); });
+
+            mavlink_message_t msg;
+            mavlink_msg_command_ack_pack(
+                _parent->get_own_system_id(),
+                _parent->get_own_component_id(),
+                &msg,
+                command.command,
+                request_ack,
+                100,
```

Comment:
```suggestion
                std::numeric_limits<uint8_t>::max(),
```

Or 255.

Only `RESULT_IN_PROGRESS` shares the progress.